Sources/ProjectAutomation/Configuration.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+// MARK: - Configuration
+
+// A the build settings of a target. It is initialized with either the `.debug` or `.release`
+
+public struct Configuration: Equatable, Codable, Comparable {
+    public enum Variant: String, Codable {
+        case debug
+        case release
+    }
+
+    public var name: String
+    public var variant: Variant
+
+    public init(
```

Comment:
What's the use case for making them public? If we don't have any, I'd suggest to make all the constructors `internal`.

Reply:
I think they are instantiated from TuistKit

Reply:
tried as much as possible but it is needed from other modules